### PR TITLE
fix: add getting changed files for PR

### PR DESCRIPTION
**Pull request by DekusDenial**
blocked by: https://github.com/screwdriver-cd/data-schema/pull/324
related: https://github.com/screwdriver-cd/screwdriver/issues/1528

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -936,20 +936,34 @@ class GithubScm extends Scm {
...
      * @return {Promise}                   Resolves to an array of filenames of the changed files
      */
-    async _getChangedFiles({ type, payload, token }) {
+    async _getChangedFiles({ type, payload, token, scmUri, prNum }) {
         if (type === 'pr') {
+            let owner;
+            let repo;
+            let number;
+
+            if (scmUri) {
```
Comment (minz1027) [inline index.js]: we should probably add a test to cover this if condition

File: index.js [hunk 2]
```diff
@@ -936,20 +936,34 @@ class GithubScm extends Scm {
      * @param  {String}   config.type      Can be 'pr' or 'repo'
      * @param  {Object}   config.payload   The webhook payload received from the SCM service.
      * @param  {String}   config.token     Service token to authenticate with Github
+     * @param  {String}   config.scmUri    The scmUri to get PR info of
```
Comment (tkyi) [inline index.js]:
```suggestion
     * @param  {String}   [config.scmUri]    The scmUri to get PR info of
```
to indicate that the param is optional

File: index.js [hunk 3]
```diff
@@ -936,20 +936,34 @@ class GithubScm extends Scm {
...
+     * @param  {Integer}  config.prNum     The PR number
```
Comment (tkyi) [inline index.js]: same here

File: index.js (hunk 1 above)
Comment (tkyi) [reply index.js]: do you need to make sure both `scmUri` && `prNum` exist?

Comment (DekusDenial) [reply index.js]: 👍

**Comment by DekusDenial** [comment]
pushed test change for the function change

**Pull request merged**